### PR TITLE
Support metadata: resolve model's metadata based on tag

### PR DESCRIPTION
**Pull request by xingjigongsi**
modify model.go

### CODE REVIEW

File: model.go [hunk 1]
```diff
@@ -1,25 +1,79 @@
...
+	if v, ok := t.metas.Load(reflect.TypeOf(table)); ok {
+		return v.(*TableMeta), nil
+	}
+	return t.Register(table)
+}
+
+func (t *tagMetaRegistry) Register(table interface{}, opts ...tableMetaOption) (*TableMeta, error) {
+	rtype := reflect.TypeOf(table)
+	v := rtype.Elem()
+	columnMeta := []*ColumnMeta{}
```
Comment (flycash) [inline model.go:52]: using `v.NumField()` as the capacity.

File: model.go [hunk 2]
```diff
@@ -1,25 +1,79 @@
...
+	fieldMap := make(map[string]*ColumnMeta)
```
Comment (flycash) [inline model.go:53]: using `v.NumField()` as the capacity

File: model.go [hunk 3]
```diff
@@ -1,25 +1,79 @@
...
+	for i := 0; i < v.NumField(); i++ {
+		dataMeta := &ColumnMeta{}
```
Comment (flycash) [inline model.go:55]: dataMeta rename to  columnMeta

File: model.go (hunk 1 above)
Comment (flycash) [reply model.go:52]: rename to `columnMetas`

File: model.go [hunk 4]
```diff
@@ -1,25 +1,79 @@
...
+		isKey := strings.Contains(tag, "primary_key")
+		dataMeta.columnName = internal.UnderscoreName(structField.Name)
+		dataMeta.fieldName = structField.Name
+		dataMeta.typ = structField.Type
+		dataMeta.isAutoIncrement = isAuto
+		dataMeta.isPrimaryKey = isKey
+		columnMeta = append(columnMeta, dataMeta)
+		fieldMap[dataMeta.fieldName] = dataMeta
+	}
+	TableMeta := &TableMeta{}
```
Comment (flycash) [inline model.go:68]:
```go
tm := &TableMeta {
    columns: columeMetas,
    // ... other fields
}
```
In general, we don't use the style 
```go
        TableMeta.columns = columnMeta
	TableMeta.tableName = internal.UnderscoreName(v.Name())
	TableMeta.typ = rtype
	TableMeta.fieldMap = fieldMap
```

Comment (flycash) [reply model.go:68]: Also, local variable's name should start with lower case character.

File: internal/commonFun.go [hunk 5]
```diff
@@ -0,0 +1,44 @@
+package internal
+
+import (
+	"reflect"
+	"unicode"
+)
+
+func TableName(table interface{}) (string, []interface{}) {
```
Comment (flycash) [inline internal/commonFun.go:8]: we don't need it anymore?

**Review by flycash — COMMENTED** [review]
do not push `delete.go`, please rollback it.

File: model.go (hunk 4 above)
Comment (xingjigongsi) [reply model.go:68]: ok

File: model.go (hunk 2 above)
Comment (xingjigongsi) [reply model.go:53]: It doesn't seem necessary

File: model.go (hunk 3 above)
Comment (xingjigongsi) [reply model.go:55]: ok

**Pull request closed without merging**